presto-main/src/main/java/com/facebook/presto/execution/CreateFunctionTask.java

```diff
@@ -95,6 +98,17 @@ private SqlInvokedFunction createSqlInvokedFunction(CreateFunction statement)
                 .build();
         String body = formatSql(statement.getBody(), Optional.empty());
 
+        if (statement.getBody() instanceof Return) {
+            Expression bodyExpression = ((Return) statement.getBody()).getExpression();
+            Type bodyType = analysis.getType(bodyExpression);
+
+            if (!bodyType.equals(metadata.getType(returnType))) {
+                // Casting is safe-here, since the check to verify if the two types are Coercible is verified as part of the statement analyzer step.
```

Comment:
nit: `... since we have verified that the actual type of the body is coercible to declared return type.`

Reply:
fixed.